app/forms/claims/claim_window_form.rb

```diff
@@ -0,0 +1,59 @@
+class Claims::ClaimWindowForm < ApplicationForm
+  attr_accessor :id
+
+  date_attribute :starts_on
+  date_attribute :ends_on
+
+  attribute :academic_year_id
+
+  validates :starts_on, presence: true
+  validates :ends_on, presence: true, comparison: { greater_than_or_equal_to: :starts_on }
+  validates :academic_year_id, presence: true
+
+  validate :does_not_start_within_another_claim_window
+  validate :does_not_end_within_another_claim_window
+
+  def initialize(attributes = {})
+    super
+
+    if id
+      self.starts_on ||= claim_window.starts_on
+      self.ends_on ||= claim_window.ends_on
+      self.academic_year_id ||= claim_window.academic_year_id
+    end
+  end
+
+  def persist
+    claim_window.update!(starts_on:, ends_on:, academic_year_id:)
+  end
+
+  def new_back_path
+    new_claims_support_claim_window_path(claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
+  end
+
+  def edit_back_path
+    edit_claims_support_claim_window_path(id, claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
+  end
+
+  def academic_year_name
+    AcademicYear.find_by(id: academic_year_id)&.name
+  end
+
+  private
+
+  def claim_window
+    @claim_window ||= id ? Claims::ClaimWindow.find(id) : Claims::ClaimWindow.new
+  end
+
+  def does_not_start_within_another_claim_window
+    return unless Claims::ClaimWindow.where.not(id:).where("starts_on <= :starts_on AND ends_on >= :starts_on", starts_on:).exists?
+
+    errors.add(:starts_on, :overlap)
+  end
+
+  def does_not_end_within_another_claim_window
+    return unless Claims::ClaimWindow.where.not(id:).where("starts_on <= :ends_on AND ends_on >= :ends_on", ends_on:).exists?
+
+    errors.add(:ends_on, :overlap)
+  end
+end
```

Comment:
```suggestion
class Claims::ClaimWindowForm < ApplicationForm
  attr_accessor :id

  date_attribute :starts_on
  date_attribute :ends_on

  attribute :academic_year_id

  validates :starts_on, presence: true
  validates :ends_on, presence: true, comparison: { greater_than_or_equal_to: :starts_on }
  validates :academic_year_id, presence: true
  validate :validate_claim_window

  def initialize(attributes = {})
    super

    if id
      self.starts_on ||= claim_window.starts_on
      self.ends_on ||= claim_window.ends_on
      self.academic_year_id ||= claim_window.academic_year_id
    end
  end

  def persist
    assign_claim_window_attributes.save!
  end

  def new_back_path
    new_claims_support_claim_window_path(claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
  end

  def edit_back_path
    edit_claims_support_claim_window_path(id, claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
  end

  def academic_year_name
    AcademicYear.find_by(id: academic_year_id)&.name
  end

  private

  def claim_window
    @claim_window ||= id ? Claims::ClaimWindow.find(id) : Claims::ClaimWindow.new
  end

  def assign_claim_window_attributes
    @assign_claim_window_attributes ||= claim_window.tap do |window|
      window.starts_on = starts_on
      window.ends_on = ends_on
      window.academic_year_id = academic_year_id
    end
  end

  def validate_claim_window
    if assign_claim_window_attributes.invalid?
      assign_claim_window_attributes.errors.each do |err|
        next if err.attribute == :academic_year

        errors.add(err.attribute, err.message)
      end
    end
  end
end
```
I've changed your code so that the validation is removed and is dependant on the validation used on the model.

Reply:
Do you mind if I revisit this in another PR? I agree with the point of leaning on model validations, as the `#persist` call would fail anyway. I'd like to give a good think about what the API can best be presented as.